packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -543,6 +548,24 @@ function commitRoot(root: FiberRoot, finishedWork: Fiber): void {
       : updateExpirationTimeBeforeCommit;
   markCommittedPriorityLevels(root, earliestRemainingTimeBeforeCommit);
 
+  let interactions = null;
+  if (enableProfilerTimer) {
+    interactions = getInteractionsForExpirationTime(
+      root,
+      committedExpirationTime,
+      true,
+    );
+
+    startContinuation(interactions !== null ? Array.from(interactions) : null);
+
+    // We store the committed interactions on the FiberRoot for two reasons:
+    // Firstly, this is how DevTools will access them when the onCommitRoot() hook is called.
+    // Secondly, this is how commitWork() will access them to pass them to any Profiler onRender() hooks.
+    // commitWork() can't use getInteractionsForExpirationTime() because the interactions have already been cleared.
+    // We can't wait to clear them out of the Map after commitWork() either, or we'll wipe out cascading updates.
+    finishedWork.stateNode.committedInteractions = interactions;
```

Comment:
You can move this and the other assignment in `createFiberRoot` to the begin phase. Also let's call it `memoizedInteractions` instead? "Committed" is misleading because it hasn't committed yet at this point. It's more like `memoizedProps` and `memoizedState`.